crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -99,6 +105,26 @@ impl Mesh {
         self.primitive_topology
     }
 
+    /// Whether this mesh has morph targets.
+    pub fn has_morph_targets(&self) -> bool {
+        self.morph_targets.is_some()
+    }
+
+    /// Set [morph targets] for this mesh using [`VisitMorphTargets`].
+    ///
+    /// [morph targets]: https://en.wikipedia.org/wiki/Morph_target_animation
+    pub fn set_morph_targets<Visit: VisitMorphTargets>(
```

Comment:
The various abstractions here (VisitMorphTargets, store_image, MorphAttributesImage) introduce too much complexity relative to their value. `store_image` fights standard dataflow in a way that indicates to me that this needs rethinking.

I've put together a refactor that I think significantly improves legibility, code size, and extensibility. It removes over 1,700 lines of code and makes this interface much more straightforward to work with. The core ideas being:
* Remove `VisitMorphTargets` in favor of iterators over `MorphAttributes`
* Turn this function "inside out". Embrace the fact that morph_targets is just an Image and allow callers to decide how that is populated. Embrace MorphTargetImage as the public interface for creating morph target images. Yes this means users could pass in any image ... but we're _already_ letting them do that with store_image ... the abstraction already leaks.
* Remove MorphAttributesImage. It seems to exist only to aid in binding lookups. We already have a simple interface for looking up bindings (`images.get(handle).map(|i| i.texture_view.clone()`). This is called exactly once ... I don't think it merits abstraction.

Reply:
Correction: only 187 lines apparently I grossly misused `git diff --stat`

Reply:
The idea behind `MorphAttributesImage` was to avoid publicly exposing the internal representation of a morph target. It's very "bytely" typed (like stringly typed but with bytes). What if we change the way each weight is laid out in the image? What if instead of an image we switch to a buffer or a more complex representation that can be spliced (for example, to handle subsetting the image to only relevant weights to reduce GPU bandwidth usage)

The `Image` is also fairly difficult to construct and tightly bound to our shader, so it becomes a perilous path to tread for the user. Though I tend to fall on the side "let the user do whatever they want"

I really like the rest of your proposed changes. I didn't know it was possible to nest `impl`s like that.